### DISCUSSION TITLE
onBlur validation with React Hook Form 6 (React 18)

### DISCUSSION THREAD

**Discussion by inPhoenix** [react-hook-form/react-hook-form#8995] [General] [answered]
Hi Everyone.
I'm trying to understand why this [live example](https://inphoenix.github.io/inreacthook/) works fine on gh-pages but not when trying to run local by a fresh create-react-app (5.0.1) installation.

```javascript
function App() {
  const onSubmit = () => {console.log('==> onSubmit')}
  const { register, errors, handleSubmit } = useForm({ mode: 'onBlur' })
  return (
      <form onSubmit={handleSubmit(onSubmit)}>
          <input name="firstName" placeholder="bill" ref={register({ required: true, maxLength: 2 })} />
          {errors.firstName && <p>"Dont leave empty or set to less than 2 chars"</p>}
      </form>
  )
}
```

So basically, in the local environment, the validation only works after I click on the checkbox.

I'm using React Hook Form 6 with React 18 deployed by create-react-app. The node version used is 16.

All code here (running by npm start)
https://github.com/inPhoenix/inreacthook
Live version (witch works):
https://inphoenix.github.io/inreacthook/

**Comment by inPhoenix** [comment, marked as answer]
I just noticed that React-Hook-Form has a dependency of React 16 | React 17

**Reply by Moshyfawn** [reply]
It does sound like a `StrictMode` issue with React 18. Try removing `StrictMode` with RHF v6 and see if it helps

**Reply by inPhoenix** [reply]
Upgrading to RHF 7 with some API changes fixed the issue.
